File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int gcd(int a, int b){
    int r;
    if(a < b){
        r = b;
        b = a;
        a = r;
    }
    r = a % b;
    if(r == 0)return b;
    
    return gcd(b, r);
}


int lcm(int a, int b){
    int g = gcd(a,b);
    return a*b/g;
}

int solution(int arr[], size_t arr_len) {
    int answer = arr[0];

    for(int i=1;i<arr_len;i++){
        answer = lcm(answer, arr[i]);
    }

    return answer;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def gcd(a, b):
    if a < b:
        a, b = b, a
    r = a % b
    if r == 0:
        return b
    return gcd(b, r)


def solution(arr):
    answer = arr[0]
    for i in range(1, len(arr)):
        g = gcd(arr[i], answer)
        answer = answer*arr[i]//g

    return answer